Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -938,36 +938,20 @@ bool MMU::IsOptimizableRAMAddress(const u32 address, const u32 access_size) cons
 }
 
 template <XCheckTLBFlag flag>
-bool MMU::IsRAMAddress(u32 address, bool translate)
+bool MMU::IsEffectiveRAMAddress(u32 address)
 {
-  if (translate)
-  {
-    auto translate_address = TranslateAddress<flag>(address);
-    if (!translate_address.Success())
-      return false;
-    address = translate_address.address;
-  }
+  return IsPhysicalRAMAddress(TranslateAddress<flag>(address).address);
```

Comment:
You still need to check what `Success` returns.

Reply:
Then I don't understand what to do with sepalani's suggestion

Reply:
I would write it like:

```
  auto translate_address = TranslateAddress<flag>(address);
  if (!translate_address.Success())
    return false;
  return IsRAMAddress<flag>(translate_address.address);
```

Sepalani's suggestion was specifically about removing the line `address = translate_address.address;`.